src/compiler/scala/tools/nsc/javac/JavaScanners.scala

```diff
@@ -702,6 +716,108 @@ trait JavaScanners extends ast.parser.ScannersCommon {
         in.next()
       }
 
+    /** read a triple-quote delimited text block, starting after the first three
+      * double quotes
+      */
+    private def getTextBlock(): Unit = {
+      // Open delimiter is followed by optional space, then a newline
+      while (isWhitespace(in.ch) && in.ch != CR && in.ch != LF && in.ch != SU) {
+        in.next()
+      }
+      if (in.ch != LF && in.ch != CR) {
+        syntaxError("illegal text block open delimiter sequence, missing line terminator")
+        return
+      }
+      in.next()
+
+      /* Do a lookahead scan over the full text block to:
+       *   - compute common white space prefix
+       *   - find the offset where the text block ends
+       */
```

Comment:
I very much dislike needing to do two passes over the input (especially in a scanner), but the alternative is to buffer up and post-process lines. Finding the offset where the block ends is not essential, but it is definitely simplifies the second pass.

Reply:
I'm not sure how we could use that. I think the Scala compiler is supposed to run in a host JVM that is 8+ even if it can target 8-16. Even then, code that depends on the host JVM version is usually hairy (since you have to detect the version and make sure it'll only ever be loaded on 16+).

Reply:
I was thinking we should re-implement it, sorry for being unclear.

Reply:
OK, that makes sense now. I see that this is [what `javac` does](https://github.com/openjdk/jdk16/blob/37043b05576c8b81b43ac41a8f06de0d0bbb3f5b/src/jdk.compiler/share/classes/com/sun/tools/javac/parser/JavaTokenizer.java#L1059) - I guess that's the benefit of being able to move the compiler and runtime forward at the same time.

Is there a reason you prefer an approach that post-processes? I opted against that because the existing machinery for parsing character escapes would need to be refactored: we need to be aware of the character escapes while collecting the initial raw string, but we have to avoid interpreting them until after we've stripped whitespace. The current `getlitch` function does the parsing and interpreting all in one step (Java uses `translateEscapes`, but that's also only available JDK 14+).

Reply:
Let me take another look with the escaping issue in mind.

My main motivation splitting it up would be to keep the code in the Java scanner simpler.

Reply:
I took another look; we could change `getlitch` to have a boolean parameter to only check or also interpret escapes. The `getTextBlock` would then be quite simple, assuming we have `translateEscapes` and `stripIdent` - which is currently not the case though. Your implementation looks good, so I agree it's the better way for now.